src/main/resources/templates/db_scripts/renameModesOfIssuance.sql

```diff
@@ -0,0 +1,23 @@
+START TRANSACTION;
+
+UPDATE ${myuniversity}_${mymodule}.mode_of_issuance AS it
+SET jsonb = jsonb_set(it.jsonb, '{name}', '"integrating resource"')
+WHERE it.jsonb->>'name' = 'Integrating Resource';
+
+UPDATE ${myuniversity}_${mymodule}.mode_of_issuance AS it
+SET jsonb = jsonb_set(it.jsonb, '{name}', '"multipart monograph"')
+WHERE it.jsonb->>'name' = 'Sequential Monograph';
+
+UPDATE ${myuniversity}_${mymodule}.mode_of_issuance AS it
+SET jsonb = jsonb_set(it.jsonb, '{name}', '"serial"')
+WHERE it.jsonb->>'name' = 'Serial';
+
+UPDATE ${myuniversity}_${mymodule}.mode_of_issuance AS it
+SET jsonb = jsonb_set(it.jsonb, '{name}', '"single unit"')
+WHERE it.jsonb->>'name' = 'Monograph';
+
+UPDATE ${myuniversity}_${mymodule}.mode_of_issuance
+SET jsonb = jsonb || '{"name": "unspecified", "source": "folio"}'
+WHERE jsonb->>'name' = 'Other';
+
+END TRANSACTION;
```

Comment:
I don't think that we need to execute update in a single transaction. Partial migration is also acceptable, since the records are not related to each other.

Reply:
@bohdan-suprun Why is partial migration acceptable? If updates to these records fails, will the module upgrade fail?

Reply:
@marcjohnson-kint, those records are not related to each other, record `A` is not depend on record `B` and the case when record `A` has been migrated but `B` hasn't, won't cause any issues.

Transaction in this context means that migration must be applied to all the records or reverted at all if at least one failed. Without using transaction all previous successful updates will be committed (i.e. in this case each update statement will be executed in it's own transaction).

>  If updates to these records fails, will the module upgrade fail?

Yes, upgrade will fail anyway, but successful updates will be kept.

Reply:
> @marcjohnson-kint, those records are not related to each other, record `A` is not depend on record `B` and the case when record `A` has been migrated but `B` hasn't, won't cause any issues.
> 
> Transaction in this context means that migration must be applied to all the records or reverted at all if at least one failed. Without using transaction all previous successful updates will be committed (i.e. in this case each update statement will be executed in it's own transaction).

What impact could a partial success / failure have on operations folks who need to back out a failed module upgrade?

> > If updates to these records fails, will the module upgrade fail?
> 
> Yes, upgrade will fail anyway, but successful updates will be kept.

As far as I am aware FOLIO has not established the expectations around partial data upgrades, based upon the conversations I've been part of, folks would prefer the data was not partially updated upon failure.

@skoczko Would that fit with your understanding?

@bohdan-suprun Is it going to be particularly expensive to use a transaction for these updates? Are you concerned about the performance overhead of that?

Reply:
> As far as I am aware FOLIO has not established the expectations around partial data upgrades, based upon the conversations I've been part of, folks would prefer the data was not partially updated upon failure.

Hmm... that's make sense to me then. We should maybe mention it somewhere - that there should be no partial upgrades. 

> @bohdan-suprun Is it going to be particularly expensive to use a transaction for these updates? Are you concerned about the performance overhead of that?

No, I don't think this affects performance, my main idea was - try to migrate as much records as we can vs revert everything if at least one update failed. 

What do you think @Svitlana-Zmiivska, @marcjohnson-kint?

Reply:
> > As far as I am aware FOLIO has not established the expectations around partial data upgrades, based upon the conversations I've been part of, folks would prefer the data was not partially updated upon failure.
> 
> Hmm... that's make sense to me then. We should maybe mention it somewhere - that there should be no partial upgrades.

I agree, FOLIO needs to agree upon that expectation and document it. I have raised it as a separate topic for the Tech Leads (the current upgrade topic has grown rather large and unwieldy)
 
> > @bohdan-suprun Is it going to be particularly expensive to use a transaction for these updates? Are you concerned about the performance overhead of that?
> 
> No, I don't think this affects performance, my main idea was - try to migrate as much records as we can vs revert everything if at least one update failed.
> 
> What do you think @Svitlana-Zmiivska, @marcjohnson-kint?

I think we should update all of them or none, if there is limited impact on performance, until we know what the correct expectations are.

Reply:
Ok, thanks @marcjohnson-kint. @Svitlana-Zmiivska, I'm resolving the conversation, no changes required in this case.